tck/src/main/java/org/eclipse/microprofile/fault/tolerance/tck/asynchronous/AsyncClassLevelClient.java

```diff
@@ -41,28 +41,16 @@
 @Asynchronous
 public class AsyncClassLevelClient {
 
-
     /**
-     * service 1 second in normal operation.
-     * @return the result as a Future
-     * @throws InterruptedException the interrupted exception
+     * Service an operation until waitCondition is completed or 1 second timeout.
+     * @param waitCondition Execution of this method will delay until the condition is finished
+     * @return the result as a Future.
      */
-    public Future<Connection> service() throws InterruptedException {
-
-        Connection conn = new Connection() {
-            {
-                Thread.sleep(1000);
-            }
-
-            @Override
-            public String getData() {
-                return "service DATA";
-            }
-        };
-
-        return CompletableFuture.completedFuture(conn);
+    @Asynchronous
+    public Future<Connection> service(Future<?> waitCondition) {
+        return serviceCS(waitCondition, false).toCompletableFuture();
```

Comment:
Yea, this is probably the easiest impl :-) But we should use the `CompletableFutureHelper` to convert `CompletionStage` to `CompletableFuture`, as `CompletionStage.toCompletableFuture()` doesn't have to be supported.

Reply:
done